File: MS/Lab 2/src/lab2/lab2/FunRand/FunRand.swift
//
//  FunRand.swift
//  lab2
//
//  Created by Andrey Meshkov on 31.12.2024.
//

import Foundation

class FunRand {
    
    private class var generateA: Double {
        var a = Double.random(in: 0..<1)
        while a == 0 {
            a = Double.random(in: 0..<1)
        }
        return a
    }
    
    class func exp(timeMean: Double) -> Double {
        var a = generateA
        a = -timeMean * log(a)
        return a
    }
    
    class func unif(timeMin: Double, timeMax: Double) -> Double {
        var a = generateA
        a = timeMin + a * (timeMax - timeMin)
        return a
    }
    
    class func norm(timeMean: Double, timeDeviation: Double) -> Double {
        timeMean + timeDeviation * Double.random(in: -Double.infinity...Double.infinity)
    }
    
}

File: Курсова робота/src2/Hospital/src/FunRand.java
public class FunRand {
    public static double Exp(double timeMean) {
        double a = 0;

        while (a == 0) {
            a = Math.random();
        }

        a = -timeMean * Math.log(a);

        return a;
    }

    public static double Unif(double timeMin, double timeMax) {
        double a = 0;

        while (a == 0) {
            a = Math.random();
        }

        a = timeMin + a * (timeMax - timeMin);

        return a;
    }

    public static double Erl(double timeMean, double k) {
        double a = 0;

        for (int i = 0; i < k; i++) {
            double u = 0;
            while (u == 0) {
                u = Math.random();
            }
            
            a += Math.log(u);;
        }

        a = -1 / timeMean * a;
        
        return a;
    }
}
